arangod/Graph/BaseOptions.cpp

```diff
@@ -605,11 +619,27 @@ void BaseOptions::isQueryKilledCallback() const {
 }
 
 void BaseOptions::setVertexProjections(Projections projections) {
+  if (!getVertexProjections().empty()) {
+    resourceMonitor().decreaseMemoryUsage(getVertexProjections().size() *
+                                          sizeof(aql::Projections::Projection));
+  }
+  arangodb::ResourceUsageScope guard(
```

Comment:
if this throws, then we will have an invalid state, because we have already subtracted the memory for the vertex projections, but the vector is not empty. To be on the safe side, the vertex projections vector must be cleared as well.

Reply:
Added in: 60044d169796e5dd8850ba2454d5f44463b1fc76